lib/moonbeam.js

```diff
@@ -80,14 +81,31 @@ class Moonbeam {
   }
 
   listen (cb = () => {}) {
-    const { port } = this.conf
+    const { port, socket } = this.conf
+    if (port && socket) throw new Error('ERR_SOCK_PORT')
 
     this.startPlugins(this._plugins, (err) => {
       if (err) return cb(err)
+
+      if (socket) return this.listenSocket(socket, cb)
+
       this.server = app.listen(port, cb)
     })
   }
 
+  listenSocket (socket, cb) {
+    fs.unlink(socket, (err) => {
+      if (err && err.code !== 'ENOENT') return cb(err)
+      this.server = app.listen(socket, (err) => {
+        if (err) return cb(err)
+        fs.chmod(socket, 0x777, (err) => {
```

Comment:
hello @robertkowalski, this should be an octal value: can you please change it to `0o777`?

Reply:
fixed, good catch!